lib/kubeclient/common.rb

```diff
@@ -99,6 +99,10 @@ def initialize_client(
       end
     end
 
+    def with_faraday_config(&block)
+      @faraday_client = create_faraday_client(&block)
+    end
+
```

Comment:
I've changed my approach following your suggestion to expose this method which takes a block that yields the `Faraday::Connection` object, which is really the only place we can specify all the middlewares and other config. Let me know if you think this is on the right track.

Reply:
nice :+1: